metrics/web/docs/features/deep_links/01_deep_linking.md

```diff
@@ -399,3 +412,214 @@ class MetricsRouteInformationParser {
 Consider this class diagram that illustrates the required changes needed to parse the deep links from the URL:
 
 ![Parsing deep links diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/master/metrics/web/docs/features/deep_links/diagrams/parsing_deep_links_class_diagram.puml)
+
+#### Applying Deep Links
+The following subsections describe the required changes needed to be able to restore the application state from the deep links and update the deep links in response to application events.
+
+##### PageParametersModel
+Once we're able to parse `query parameters`, we should have a way to represent them in the Metrics Web application. To do that, let's introduce a `PageParametersModel`. 
+
+The `PageParametersModel` is an interface and does not store any data, however, it creates a general interface for all implementers including the `toMap() : Map<String, dynamic` method. 
+
+The implementers of the `PageParametersModel` should include the parameters that are specific to a `page` they relate to, and implement the `PageParametersModel` interface.
+
+The following class diagram demonstrates the general approach for creating new `PageParametersModel`s on a `SomePageParametersModel` example:
```

Comment:
:readable:
It seems like an interface could not store any data itself.
```suggestion
The `PageParametersModel` is an interface for models that stores the application state parameters parsed from the query parameters. This interface provides the `.toMap()` method that the application uses to serialize appropriate models to the query parameters `Map`. The implementers of the `PageParametersModel` should include the parameters that are specific to a `page` they relate to. Also, they should know how to serialize and deserialize their data using the `.toMap()` method and `.fromMap()` constructor respectively. 

The following class diagram demonstrates the general approach for creating new `PageParametersModel`s on a `SomePageParametersModel` example:
```

Also, what do you think about renaming the `SomePageParametersModel` to `ConcretePageParametersModel` (to be more strict here) or `CoolPageParametersModel` (as in metrics web architecture documents).